x-pack/plugins/ml/server/routes/data_frame_analytics.ts

```diff
@@ -13,12 +14,49 @@ import {
   dataAnalyticsExplainSchema,
   analyticsIdSchema,
   stopsDataFrameAnalyticsJobQuerySchema,
+  deleteDataFrameAnalyticsJobSchema,
 } from './schemas/data_analytics_schema';
+import { IndexPatternHandler } from '../models/data_frame_analytics/index_patterns';
+
+export interface DeleteDataFrameAnalyticsWithIndexStatus {
+  success: boolean;
+  error?: CustomHttpResponseOptions<ResponseError>;
+}
+
+function getIndexPatternId(context: RequestHandlerContext, patternName: string) {
+  const iph = new IndexPatternHandler(context.core.savedObjects.client);
+  return iph.getIndexPatternId(patternName);
+}
+
+function deleteIndexPatternById(context: RequestHandlerContext, indexPatternId: string) {
+  const iph = new IndexPatternHandler(context.core.savedObjects.client);
+  return iph.deleteIndexPatternById(indexPatternId);
+}
 
 /**
  * Routes for the data frame analytics
  */
 export function dataFrameAnalyticsRoutes({ router, mlLicense }: RouteInitialization) {
+  async function userCanDeleteIndex(
+    context: RequestHandlerContext,
+    destinationIndex: string
+  ): Promise<boolean> {
+    if (!mlLicense.isSecurityEnabled()) {
+      return true;
+    }
+    const privilege = await context.ml!.mlClient.callAsCurrentUser('ml.privilegeCheck', {
+      body: {
+        index: [
+          {
+            names: [destinationIndex], // uses wildcard
+            privileges: ['delete_index'],
+          },
+        ],
+      },
+    });
+    return privilege.index[destinationIndex].delete_index === true;
```

Comment:
Will `privilege.index[destinationIndex].delete_index` always be defined? Might be worth ensuring `privilege` and `privilege.index[destinationIndex]` are defined to avoid a possible TypeError if one of these is undefined.

Reply:
`privilege.has_all_requested` can be used here too

Reply:
Done in https://github.com/elastic/kibana/pull/66934/commits/5bd97e9c466ad1a88d32da4f432583d5ddc3aed0

Reply:
this change has also introduced a check for `privilege.securityDisabled === true ` which doesn't exist on the server side.